src/button/IconButton.tsx

```diff
@@ -40,6 +39,7 @@ export const IconButton = forwardRefWithAs<
     ? React.cloneElement(children, {
         "aria-hidden": true,
         focusable: false,
+        role: "img",
```

Comment:
Added Role as `img` - https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Role_Img#svg_and_roleimg

Reply:
Got this while surfing React Spectrum codebase.